honeybee/room.py

```diff
@@ -558,11 +575,16 @@ def solve_adjacency(rooms, tolerance):
                             if not isinstance(face_2.boundary_condition, Surface):
                                 if face_1.geometry.is_centered_adjacent(
                                         face_2.geometry, tolerance):
-                                    face_1.set_adjacency(face_2)
```

Comment:
I see what are you trying to do here and from your explanation it is about giving better access to the objects but I feel there should be a better approach to do this.

To me it looks much cleaner to set the adjacency as you have it here. I would say one can access the objects and change construction, etc. afterwards.

Reply:
Figuring out which adjacencies were solved for in the run of `solve_adjacnecy` and which ones were already existing before the run isn't easy without some output for this method. So I'm afraid the alternative you propose here isn't really usable without some major caveats. Here are some of the things I considered as an alternative to what is implemented here:

1. In the Grasshopper component, just don't use this staticmethod and write my own method to solve adjacency inside the component that sets custom constructions and face types at the time when we are setting adjacency between faces. I think you'd agree that this is not good and it's much better to have a method in the Python libraries that we actually use.

2. In the Grasshopper component, loop through all of the Rooms before running `solve_adjacnecy` to find Faces that already have an adjacent Face. Then run this `solve_adjacnecy` method and again loop through all of the Faces in an attempt to determine which ones are new. Assign custom constructions to just the new ones. I think you will agree that the current implementation of just returning the new adjacency information is much cleaner than having to loop through the data twice and attempt to figure out what happened.

3. Add inputs to this `solve_adjacency` staticmethod for custom interior radiance materials and constructions, even though this is the core library and it is supposed to be agnostic towards particular extensions. For this, I'd have to add a try/except loop in the staticmethod that handles the case that a particular extension is not installed. I think you'd agree that this is in violation of our vision to leave extension-specific stuff out of the core library. If we make an exception here for particular extensions, then people might make an argument to include code about their own extensions down the line. In the words of Python Zen "Special cases aren't special enough to break the rules."

If you think any of these proposals is better than what I have implemented, let me know. Also, let me know if there's another alternative that I am not seeing.